### PR TITLE
Profile upload feature

### PR DESCRIPTION
**Pull request by patel1643**
- added profile picture upload along with some misc. fixed

### CODE REVIEW

**Review by Mwpereira — CHANGES_REQUESTED** [review]
image

Please allow only jpg, jpeg, and png if possible both explore upload item and profile image item

File: client/src/components/SideNavBar.tsx [hunk 1]
```diff
@@ -74,7 +74,7 @@ const SidebarContent = ({ onClick }: { onClick: MouseEventHandler }) => {
             const response: any = await logout();
             console.log(response.data);
```
Comment (Mwpereira) [inline client/src/components/SideNavBar.tsx:75]: remove console.log please

File: client/src/components/UploadModal.tsx [hunk 2]
```diff
@@ -57,28 +57,29 @@ export const UploadModal = () => {
             title: e.target.title.value,
         };
 
-        if(formData.format == '') {
+        if (formData.format == '') {
```
Comment (Mwpereira) [inline client/src/components/UploadModal.tsx:60]: Please use `===`

File: client/src/components/SideNavBar.tsx (hunk 1 above)
Comment (patel1643) [reply client/src/components/SideNavBar.tsx:75]: my b

File: client/src/components/UploadModal.tsx [hunk 3]
```diff
@@ -57,28 +57,29 @@ export const UploadModal = () => {
...
             toast({
                 duration: 5000,
                 isClosable: false,
                 status: 'error',
                 title: 'Select an image to upload',
             });
-        }else if(formData.title == ''){
+        } else if (formData.title == '') {
```
Comment (Mwpereira) [inline client/src/components/UploadModal.tsx:67]: Please use `===`

File: client/src/components/UploadModal.tsx [hunk 4]
```diff
@@ -57,28 +57,29 @@ export const UploadModal = () => {
...
             toast({
                 duration: 5000,
                 isClosable: false,
                 status: 'error',
                 title: 'Enter a title for the image',
             });
-        }else if(formData.caption == ''){
+        } else if (formData.caption == '') {
```
Comment (Mwpereira) [inline client/src/components/UploadModal.tsx:74]: Please use `===`

Comment (patel1643) [reply client/src/components/UploadModal.tsx:74]: lol these aren't my changes but I will update it

File: client/src/components/UploadModal.tsx (hunk 3 above)
Comment (patel1643) [reply client/src/components/UploadModal.tsx:67]: lol these aren't my changes but I will update it

File: client/src/components/UploadModal.tsx (hunk 2 above)
Comment (patel1643) [reply client/src/components/UploadModal.tsx:60]: lol these aren't my changes but I will update it

File: client/src/components/UploadModal.tsx (hunk 4 above)
Comment (Mwpereira) [reply client/src/components/UploadModal.tsx:74]: My bad, no worries 🔥

File: client/src/components/UploadModal.tsx (hunk 3 above)
Comment (Mwpereira) [reply client/src/components/UploadModal.tsx:67]: My bad, no worries 🔥

File: client/src/components/UploadModal.tsx (hunk 2 above)
Comment (Mwpereira) [reply client/src/components/UploadModal.tsx:60]: My bad, no worries 🔥

File: client/src/components/SideNavBar.tsx (hunk 1 above)
Comment (Mwpereira) [reply client/src/components/SideNavBar.tsx:75]: No problem sir

**Review by Mwpereira — APPROVED** [review]
Everything LGTM & works perfectly on my end. Thanks for the enormous effort and work.  (Please make the minor change to which formats you can upload).

**Pull request merged**